src/core/packages/saved-objects/migration-server-internal/src/actions/catch_retryable_es_client_errors.ts

```diff
@@ -9,18 +9,30 @@
 
 import * as Either from 'fp-ts/Either';
 import type { errors as EsErrors } from '@elastic/elasticsearch';
-import { isRetryableEsClientError } from '@kbn/core-elasticsearch-server-internal';
+import { isRetryableEsClientError } from '@kbn/core-elasticsearch-server-utils';
 
 export interface RetryableEsClientError {
   type: 'retryable_es_client_error';
   message: string;
   error?: Error;
 }
 
+// Migrations also retry on Auth exceptions as this is a common failure for newly created
+// clusters that might have misconfigured credentials.
+const retryResponseStatuses = [
+  401, // AuthorizationException
+  403, // AuthenticationException
```

Comment:
The shared function doesn't retry on auth errors since usually it's not a transient error.

Reply:
I wonder if we should populate the ones below by spreading the default const `...DEFAULT_RETRY_STATUS_CODES`

Reply:
Another option would be to use an incremental approach:

```
{
  alsoRetryOn: [401, 403],
  doNotRetryOn: [xxx, yyy],
}
```

this way users can automatically benefit from updates in the DEFAULT_RETRY_STATUS_CODES list, and they can add their custom exceptions.